app/src/main/java/org/mozilla/reference/browser/library/bookmarks/ui/BookmarkViewModel.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.mozilla.reference.browser.library.bookmarks.ui
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import mozilla.components.concept.storage.BookmarkNode
+import mozilla.components.concept.storage.SearchResult
+import org.mozilla.reference.browser.library.bookmarks.usecases.BookmarkUseCases
+
+class BookmarkViewModel(private val bookmarkUseCases: BookmarkUseCases) : ViewModel() {
+
+    var bookmarkList = listOf<BookmarkNode>()
```

Comment:
Why do you need this property?

Reply:
this was done badly. I updated the implementation. It doesn't expose the mutablelivedata now and has a different field which is exposed https://gist.github.com/humblehacker/0eb6458b1df6cf3049e031f36f0615f5